packages/orbit-components/src/Stack/index.tsx

```diff
@@ -132,17 +139,19 @@ const Stack = (props: Props) => {
 
     return cx(
       typeof spaceAfter !== "undefined" && getSpaceAfterClasses(spaceAfter, viewport),
-      typeof align !== "undefined" && getAlignItemsClasses(align, viewport),
-      typeof align !== "undefined" && getAlignContentClasses(align, viewport),
-      typeof wrap !== "undefined" && getWrapClasses(wrap, viewport),
-      typeof grow !== "undefined" && getGrowClasses(grow, viewport),
-      typeof shrink !== "undefined" && getShrinkClasses(shrink, viewport),
-      typeof justify !== "undefined" && getJustifyClasses(justify, viewport),
-      getDirectionClasses(direction, viewport),
+      (flex || inline) && [
+        flex && "flex",
```

Comment:
also, it's a good candidate for adding visual tests for chromatic. Probably the best one.

Reply:
That can't be like that either. If one adds the prop `flex`, it will have both `flex` and `block` classes, which are incompatible.

My suggestion was covering that.

If `inline || flex`:
- If `true`, check if inline:
  - if `true`, apply `inline-flex`
  - if `!true`, apply `flex`
  - Apply all the other classes normally, that we currently have inside the array
- If `false`, apply `block`

This would go with something like

```js
flex || inline
  ? [
      inline ? "inline-flex" : "flex",
      typeof align !== "undefined" && getAlignItemsClasses(align, viewport),
      typeof align !== "undefined" && getAlignContentClasses(align, viewport),
      typeof wrap !== "undefined" && getWrapClasses(wrap, viewport),
      typeof grow !== "undefined" && getGrowClasses(grow, viewport),
      typeof shrink !== "undefined" && getShrinkClasses(shrink, viewport),
      typeof justify !== "undefined" && getJustifyClasses(justify, viewport),
      getDirectionClasses(direction, viewport),
    ]
  : "block",
```

**However**, while writing this, I also spotted that we are not taking the flex prop into consideration for viewports (never were)… So it seems to require more refactor 🤔 

IMO, the following would work:

Have a `getDisplayClass(flex, inline, viewport)` and inside the function get the correct class based on the `flex` and `inline` arguments:
- If `inline` => class `inline-flex`
- If `flex` => class `flex`
- else => class `block`
with the corresponding viewport prefixes, as the object already does.

After having this, we unconditionally call this function.